File: C/number_guessing_game.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {

    int lowest_num = 1;
    int highest_num = 100;
    int answer, guess;
    int guesses = 0;
    int is_running = 1;

    //random number
    srand(time(0));
    answer = (rand() % (highest_num - lowest_num + 1)) + lowest_num;

    printf("Welcome to number guessing game in C!\n");
    printf("I selected a number between %d to %d. Take a guess.\n\n",
           lowest_num, highest_num);

    while (is_running) {

        printf("Please enter the number between %d to %d: ",
               lowest_num, highest_num);

        
        if (scanf("%d", &guess) != 1) {
            
            while (getchar() != '\n');
            printf("Not a valid number. Please enter digits only!\n\n");
            continue;
        }

        guesses++;

        if (guess < lowest_num || guess > highest_num) {
            printf("Guess is out of range.\n\n");
            continue;
        }

        if (guess < answer) {
            printf("The guess is too LOW!!\n\n");
        }
        else if (guess > answer) {
            printf("The guess is too HIGH!!\n\n");
        }
        else {
            printf("\nYES!! YOU CRACKED IT.\n");
            printf("You took %d guesses to find the number %d.\n",
                   guesses, answer);
            is_running = 0;
        }
    }

    return 0;
}

File: Python/number_guessing_game.py
#let us build random number guessing game in python
import random

lowest_num=1
highest_num=100
answer=random.randint(lowest_num,highest_num)
guesses=0
is_running=True

print("Welcome to number guessing game in python " )
print(f"I selected a number between {lowest_num} to {highest_num} take a guess.")


while is_running:

    guess=input(f"Please enter the number b/w {lowest_num} to {highest_num}:")

    if guess.isdigit():
        guess=int(guess)
        guesses+=1
        if guess<lowest_num or guess>highest_num:
            print("Guess is out of range")
            guess=input(f"Please enter the number b/w {lowest_num} to {highest_num}:")
        elif guess<answer :
            print("The guess is too low!!")
        elif guess>answer:
            print("The guess is too high!!")
        else:
            print("YES!! YOU CRACK IT.")
            is_running=False
            print(f"The number of guesses you took is {guesses} to guess the number {answer}.")

    else:
        print("Not a valid guess")
        print(f"please select a number between {lowest_num} to {highest_num}")